projectX/projectX/Utilities/Extensions.swift

```diff
@@ -136,3 +137,45 @@ func withBoldText(text: String, font: UIFont? = nil) -> NSAttributedString {
   fullString.addAttributes(boldFontAttribute, range: range)
   return fullString
 }}
+
+extension Date {
+    
+  // Function in which calculates the time past on a particular post and returns
+  // this time as a string
+  func diff(date: Date) -> String {
```

Comment:
Because the class you're extending is the same as the argument type you're passing to the function, a cleaner solution would be to remove the argument and use self and it would get the exact results but with a cleaner way to call the function. This is what that would look like in this case:

```
func diff() -> String {
        
        let date = self 
```
...
and a call to this would look like:
`Date(timeIntervalSinceReferenceDate: 123456789.0).diff()`

Vs what it would look like currenttly:
`Date(timeIntervalSinceReferenceDate: 123456789.0).diff(date: Date(timeIntervalSinceReferenceDate: 123456789.0))`

Reply:
also because of how long the date object is here the effect is a bit exaggerated but hopefully you see what i mean